cli/src/action/workload.rs

```diff
@@ -170,30 +168,21 @@ impl Action for WorkloadAction {
             }
         }
 
-        // calculate the end time based on the duration given
-        let end_time = duration.map(|d| Instant::now() + Duration::from(d));
-
         // setup control-c handling
-        let running = Arc::new(AtomicBool::new(true));
-        let r = running.clone();
+        let workload_runner_shutdown_signaler = workload_runner.shutdown_signaler();
```

Comment:
A better way to do this is to use a simple channel, in the ctrlc handler and then just shutdown the workload runner, without the extra signaller. For example:

```rust
        let (shutdown_tx, shutdown_rx) = std::sync:mpsc::channel();
        ctrlc::set_handler(move || {
            if let Err(err) = shutdown_tx.send(()) {
                // already called control-c, so process exit
            }
        })
        .map_err(|_| {
            CliError::ActionError("Unable to set up workload ctrlc handler".to_string())
        })?;

        let _ = shutdown_rx.recv().ok_or(());
        drop(shutdown_rx);
        workload_runner.shutdown(); // signal and wait, omitting error handling for breviety 
```

Reply:
I think this would create a similar issue as before where if there is an error in the loop in `fn build()` it will break the loop and stop submitting workloads but the program will stall on this line `let _ = shutdown_rx.recv().ok_or(());` and not exit until a ctrl-c is received.